src/main/java/me/jellysquid/mods/lithium/mixin/world/explosions/ExplosionMixin.java

```diff
@@ -220,44 +232,47 @@ private float traverseBlock(float strength, int blockX, int blockY, int blockZ,
-
-                // If the block state is air, it cannot have fluid or any kind of resistance, so just leave
-                if (blockState.getBlock() != Blocks.AIR) {
-                    // Rather than query the fluid state from the container as we just did with the block state, we can
-                    // simply ask the block state we retrieved what fluid it has. This is exactly what the call would
-                    // do anyways, except that it would have to retrieve the block state a second time, adding overhead.
-                    FluidState fluidState = blockState.getFluidState();
-
-                    // Pick the highest resistance value between the block and fluid
-                    float resistance = Math.max(blockState.getBlock().getBlastResistance(), fluidState.getBlastResistance());
-
-                    // If this explosion was caused by an entity, allow for it to modify the resistance of this position
-                    if (this.entity != null) {
-                        resistance = this.entity.getEffectiveExplosionResistance((Explosion) (Object) this, this.world, pos, blockState, fluidState, resistance);
+        float totalResistance = 0.0F;
+        Optional<Float> blastResistance;
+
+        labelGetBlastResistance:
+        {
+            // If the chunk is missing or out of bounds, assume that it is air
+            if (chunk != null) {
+                // We operate directly on chunk sections to avoid interacting with BlockPos and to squeeze out as much
+                // performance as possible here
+                ChunkSection section = chunk.getSectionArray()[blockY >> 4];
+
+                // If the section doesn't exist or it's empty, assume that the block is air
+                if (section != null && !section.isEmpty()) {
+                    // Retrieve the block state from the chunk section directly to avoid associated overhead
+                    blockState = section.getBlockState(blockX & 15, blockY & 15, blockZ & 15);
+
+                    // If the block state is air, it cannot have fluid or any kind of resistance, so just leave
+                    if (blockState.getBlock() != Blocks.AIR) {
+                        // Rather than query the fluid state from the container as we just did with the block state, we can
+                        // simply ask the block state we retrieved what fluid it has. This is exactly what the call would
+                        // do anyways, except that it would have to retrieve the block state a second time, adding overhead.
+                        FluidState fluidState = blockState.getFluidState();
+
+                        // Get the explosion resistance like vanilla
+                        blastResistance = this.behavior.getBlastResistance((Explosion) (Object) this, this.world, pos, blockState, fluidState);
+                        break labelGetBlastResistance;
```

Comment:
Possibly cleaner to have this be a separate function to avoid the label and break